### PR TITLE
Added route to get NEM's metadata by composite hash

### PR DESCRIPTION
**Pull request by xgreenx**

### CODE REVIEW

File: sdk/metadata_nem.go [hunk 1]
```diff
@@ -1,12 +1,65 @@
...
+	resp, err := ref.client.doNewRequest(ctx, http.MethodPost, metadataEntriesRoute, &computedHashes{hashes}, &dtos)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = handleResponseStatusCode(resp, map[int]error{400: ErrInvalidRequest, 409: ErrArgumentNotValid}); err != nil {
+		return nil, err
+	}
+
+	mscInfos, err := dtos.toStruct(ref.client.config.NetworkType)
```
Comment (jenyasd209) [inline sdk/metadata_nem.go:56]: why don't you just return `dtos.toStruct(ref.client.config.NetworkType)`?

File: sdk/metadata_nem_internal.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	"encoding/hex"
+	jsonLib "encoding/json"
+	"errors"
+)
+
+type metadataNemInfoDTOs []*metadataNemInfoDTO
+
+func (ref *metadataNemInfoDTOs) toStruct(networkType NetworkType) ([]*MetadataNemTupleInfo, error) {
+	var (
+		dtos  = *ref
```
[the remark this answers is not in the document]
Comment (xgreenx) [reply sdk/metadata_nem_internal.go:13]: Just copied code from another our implementation)

File: sdk/metadata_nem_internal.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+			return nil, err
+		}
+
+		metadataInfo.Mosaic = &MosaicMetadataNemInfo{
+			MetadataNemInfo: commonMetadata,
+			MosaicId:        mosaicId,
+		}
+
+		return metadataInfo, nil
+	} else {
```
[the remark this answers is not in the document]
Comment (xgreenx) [reply sdk/metadata_nem_internal.go:113]: Yes, but I'm too lazy to change the code)

File: sdk/metadata_nem_internal.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+func (ref *computedHashes) MarshalJSON() ([]byte, error) {
+	buf := []byte(`{"compositeHashes": [`)
+
+	for i, nsId := range ref.Hashes {
+		if i > 0 {
+			buf = append(buf, ',')
+		}
+
+		buf = append(buf, []byte(`"`+nsId.String()+`"`)...)
```
[the remark this answers is not in the document]
Comment (xgreenx) [reply sdk/metadata_nem_internal.go:130]: No, we don't know the size of each id)

**Pull request merged**